url8d1df200a8_gilleain_signatures_tgz-pJ8-signature_chemistry_LargeMoleculeTestJ8/src/signature/chemistry/MoleculeWriter.java

```diff
@@ -0,0 +1,71 @@
+package signature.chemistry;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+
+public class MoleculeWriter {
+    
+    public static void writeMolfile(String filename, Molecule molecule) {
+        File file = new File(filename);
+        try {
+            writeToStream(new FileOutputStream(file), molecule);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the resource created by the following constructor: *FileOutputStream*. The resource is referenced at line 16. The resource is closed at line 35. There are other execution paths that do not contain closure statements, for example, when BufferedWriter.newLine (in method writeHeader) throws an exception. Ensure the object returned by *FileOutputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).